torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -431,6 +438,13 @@ def __init__(
         self.process_group = process_group or _get_default_group()
         self.rank = self.process_group.rank()
         self.world_size = self.process_group.size()
+
+        self.device_id = device_id
+        if self.device_id is not None:
+            # Move module to device specified. Note that this is done prior to
+            # setting compute_device to ensure that they align.
+            module = module.to(self.device_id)
```

Comment:
check: if module is a GPU module, the GPU module's device is the same as device_id, and also similar to DDP, check the GPU module was placed in the single device

Reply:
Sure, will add this. But this is different from DDP right, DDP supports multi-device modules (it is how DDP(Pipe()) works?)

Reply:
> DDP supports multi-device modules (it is how DDP(Pipe()) works?)

Yep. I am wondering if FSDP might also need to support similar multi-device in the future (low-pri for now)? At least that could work with FSDP without param sharding?

Reply:
For now we'll assume multi-device support is not there because we haven't actively tested / worked on this?